linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -21,245 +21,254 @@
   ansible.builtin.set_fact:
     src_iso_file_name: "{{ datastore_file_download_result.dest | basename }}"
+    src_iso_file_dir: "{{ datastore_file_download_result.dest | dirname }}"
     src_iso_file_path: "{{ datastore_file_download_result.dest }}"
 
-# Get boot images from source ISO
-- include_tasks: get_boot_images_from_iso.yml
-
-# Create a temporary directory to mount ISO image
-- name: "Create a temporary directory to mount ISO image"
-  include_tasks: ../../common/create_temp_file_dir.yml
-  vars:
-    tmp_dir: "{{ unattend_iso_cache }}"
-    tmp_state: "directory"
-    tmp_prefix: "expand_{{ src_iso_file_name | replace('.iso', '') }}_"
-
-- name: "Set fact of the temp dir to mount OS ISO image"
-  ansible.builtin.set_fact:
-    tmp_iso_mount_dir: "{{ tmp_path }}"
-
-# Mount original ISO image at local
-- include_tasks: ../../common/local_mount.yml
-  vars:
-    mount_path: "{{ tmp_iso_mount_dir }}"
-    mount_src: "{{ src_iso_file_path }}"
-    mount_opts: "ro,noauto"
-    mount_fstype: "iso9660"
-
 - name: "Set fact of unattend install config file name"
   ansible.builtin.set_fact:
     unattend_install_file_name: "{{ new_unattend_install_conf | basename }}"
 
-- name: "Initialize fact of command to rebuild unattended install ISO"
-  ansible.builtin.set_fact:
-    rebuild_iso_cmd: "xorriso -indev {{ src_iso_file_path }} -outdev {{ rebuilt_unattend_iso_path }}"
-
-# Set facts about files and command for creating an autoinstall ISO for Photon
 - block:
-    # files for autoinstall in BIOS/UEFI mode
-    - name: "Set fact of files to be update for unattended install"
-      ansible.builtin.set_fact:
-        iso_files_to_update:
-          - '/boot/grub2/grub.cfg'
-          - '/isolinux/isolinux.cfg'
-          - '/isolinux/menu.cfg'
+    - name: Extract specific files inside ISO
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - "boot/grub/grub.cfg"
+    
+    - name: Modify boot entry
+      ansible.builtin.replace:
+        path: "{{ src_iso_file_dir }}/grub.cfg"
+        regexp: "set timeout=[1-9][0-9]{0,1}"
+        replace: "default=0\nset timeout=2"
 
-    - name: "Add command options to rebuild unattended install ISO for Debian"
-      ansible.builtin.set_fact:
-        rebuild_iso_cmd: "{{ rebuild_iso_cmd }} -map {{ unattend_iso_cache }}/{{ item | basename }} '{{ item }}'"
-      with_items: "{{ iso_files_to_update | union(['/isolinux/ks.cfg'])}}"
-  when: unattend_install_conf is match('Photon')
+    - name: Modify boot options
+      ansible.builtin.replace:
+        path: "{{ src_iso_file_dir }}/grub.cfg"
+        regexp: "file=/cdrom/preseed/ubuntu.seed maybe-ubiquity quiet splash ---"
+        replace: "file=/cdrom/preseed/ubuntu.seed boot=casper debug-ubiquity automatic-ubiquity quiet splash noprompt --- console=ttyS0,115200n8"
+
+    - name: "Check the content of modified file"
```

Comment:
It is more proper to use "Print" instead of "Check" here.

Reply:
Thanks. Done